opencti-platform/opencti-graphql/src/modules/attributes/internalObject-registrationAttributes.ts

```diff
@@ -240,6 +240,7 @@ const internalObjectsAttributes: { [k: string]: Array<AttributeDefinition> } = {
       multiple: false,
       upsert: false,
       isFilterable: false,
+      sortBy: 'max_confidence',
```

Comment:
To enable sorting on complex object, I suggest we enhance the schema definition, to have this optional field `sortBy` that should point to one of the field in the mapping.

This implementation here allows only for a first-level field, but it might be improved with a dot-path for instance. In the elastic search, it ends up as a dot-path anyway.

Reply:
cc @richard-julien WDYT?

Reply:
Honestly i would prefer directly the support of user_confidence_level.max_confidence in sorting field.
Any problem or specific difficulty to allow use path notation directly in the sort field asked by the user?

Reply:
If the approach is OK, I can go for the full path in the sortBy field, yes.

Note that one the client side, we manage sort-by using gql enums, and not a string variable. Thus we have fixed behaviors. It fits our mostly-flat model, but with complex object it's not a good fit.

A string variable with path notation compatible with all kind of attributes would be nice too (a bit like we have now with filters, but with sorting too). Not for this PR though 😇

Reply:
To sum up our discussions: we'll go for a full path like we do with the patching API `object_path` param.

`sortBy` should be an array of objects like `{ key: 'confidence', path: 'group_confidence_level/max_confidence' }`
Having an array will allow for several possible sorting strategy for a single complex object.

This way, the API is future-proof (as far as we know today). The actual implementation will still be a first-level approach like we did first in this PR. When (if) we need to go further into that direction we'll know how to do it but such complexity is unnecessary for now.

Reply:
We also could use some refactoring of the schema utilities: the isXXXAttribute functions are just checking the top-level attribute definition but the proper way is to accept an attribute def as parameter, so we can use these utility functions on inner mappings.